airflow/executors/fargate_executor.py

```diff
@@ -0,0 +1,327 @@
+        describe_dict = self.ecs.describe_tasks(tasks=all_task_arns, cluster=self.cluster)
+        self.log.debug('ECS Response: %s' % describe_dict)
+        # check & handle the failures first in the JSON response.
+        if describe_dict['failures']:
+            self._sync_api_failures(describe_dict['failures'])
+        # check & handle airflow sucesses & failures next
+        updated_tasks = describe_dict['tasks']
+        for task_json in updated_tasks:
+            task = FargateTask(task_json)
+            # Update the active workers with new statuses. Consequtive task failure counter resets back to 0.
+            self.active_workers.update_task(task)
+            # get state of current task
+            task_state = task.get_task_state()
+            task_key = self.active_workers.arn_to_key[task.task_arn]
+            # mark finished tasks as either a success/failure
+            if task_state == State.FAILED:
+                self.fail(task_key)
+            elif task_state == State.SUCCESS:
+                self.success(task_key)
+            # if task is no longer running then remove from active workers and place in finished
+            if task_state != State.RUNNING:
+                self.log.debug('Task %s marked as %s after running on %s' % (task_key, task_state, task.task_arn))
+                self.active_workers.pop_by_key(task_key)
+
+    def _sync_api_failures(self, arn_failures):
+        """
+        In the case of an API failure this executor will try a set amount of retries. Sometimes
+        one call to AWS might fail intermittently, and just needs to be called again in the next heartbeat. If
+        a given API call fails consecutively to a 'MAX_FAILURE_CHECKS' number of calls, the task is marked as failed.
+        """
+        for failure in arn_failures:
+            arn = failure["arn"]
+            task = self.active_workers.task_by_arn(arn)
+            task.api_failure_count += 1
+            # Sometimes ECS doesn't update the ARN key right away, and will show up as missing. Check later.
+            if task.api_failure_count >= self.__class__.MAX_FAILURE_CHECKS:
+                task_key = self.active_workers.arn_to_key[arn]
+                self.active_workers.pop_by_key(task_key)
+                self.log.error('ECS Executor could not find task %s because %s; '
+                               'marking key %s as failed' % (arn, failure["reason"], task_key))
+                self.fail(task_key)
+            else:
+                self.log.debug('ECS Executor could not find task %s because %s; '
+                               'skipping for now on strike #%s' % (arn, failure["reason"], task.api_failure_count))
+
+    def execute_async(self, key: TaskInstanceKeyType, command: CommandType,
+                      queue=None, executor_config=None):
+        """
+        This method will execute the command asynchronously.
+        """
+        # run a task
```

Comment:
I don't see a lot value in this comment.

Reply:
I can't argue with that